### ISSUE TITLE
Unexpected "unreachable expression" warning in function with a loop

### ISSUE THREAD

**Issue by nicholasbishop** [withoutboats/fehler#52]
Example code:

```rust
use fehler::throws;
use std::io::Error;

fn no_warning() -> Result<(), Error> {
    loop {
        return Ok(());
    }
}

#[throws]
fn unexpected_warning() {
    loop {
        return;
    }
}

fn main() {
    no_warning().unwrap();
    unexpected_warning().unwrap();
}
```

Gives this warning:

```
warning: unreachable expression
  --> src/main.rs:10:1
   |
10 |   #[throws]
   |   ^^^^^^^^^ unreachable expression
11 |   fn unexpected_warning() {
12 | /     loop {
13 | |         return;
14 | |     }
   | |_____- any code following this expression is unreachable
   |
   = note: `#[warn(unreachable_code)]` on by default
   = note: this warning originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

```
$ rustc --version
rustc 1.43.0 (4fb7144ed 2020-04-20)
```

**Comment by withoutboats** [comment]
Can you build against git master of fehler and confirm if this warning is still given?

**Comment by nicholasbishop** [comment]
Looks like it's fixed in master, specifically in df2e3d43aafc3966f0789d678975ebf2011bb10b. Thanks :)
